libs/networking/http_helpers/http_helpers.ml

```diff
@@ -35,176 +43,115 @@ let client_ref : (module Cohttp_lwt.S.Client) option ref = ref None
-     if it isn't on port 80/443 *)
-  (* So HTTP_PROXY=http://localhost:8080 or
-     HTTPS_PROXY=https://localhost:8080 *)
-  (* And note that HTTP_PROXY is only used for HTTP requests! We almost
-     exclusively use HTTPS urls so make sure both are set*)
-  let get_proxy uri =
-    let proxy_uri_env =
-      match Uri.scheme uri with
-      | Some "http" -> Some "HTTP_PROXY"
-      | Some "https" -> Some "HTTPS_PROXY"
-      | _ -> None
-    in
-    let proxy_uri = Option.bind proxy_uri_env Sys.getenv_opt in
-    match proxy_uri with
-    | Some proxy_uri ->
-        Log.info (fun m -> m "Using proxy %s for request" proxy_uri);
-        Some (Uri.of_string proxy_uri)
-    | None -> None
-
-  (* Why this wrapper function? Client.call takes a uri, and some other things *)
-  (* and then makes a Request.t with said uri and sends that request to the same uri *)
-  (* By using Client.callv, we can make a request that has some uri, but then really *)
-  (* send it to a different uri. This is used for proxying requests *)
-
-  let default_resp_handler (response, body) =
-    let%lwt body_str = Cohttp_lwt.Body.to_string body in
-    Lwt.return (response, body_str)
-
-  (* Why do we need a response_handler? From the cohttp docs: *)
-  (*
+let server_response_of_response (response, body) meth =
+  let code = response |> Response.status |> Code.code_of_status in
+  let meth_str = string_of_meth meth in
+  match code with
+  | _ when Code.is_success code -> { body = Ok body; response; code }
+  | _ when Code.is_error code ->
+      Log.debug (fun m -> m "HTTP %s failed:\n %s" meth_str body);
+      { body = Error body; response; code }
+  | _ ->
+      Log.debug (fun m -> m "HTTP %s unexpected response:\n %s" meth_str body);
+      { body = Error body; response; code }
+
+(*****************************************************************************)
+(* Proxy Stuff *)
+(*****************************************************************************)
+
+(* Respect the proxy!*)
+(* Cohttp doesn't https://github.com/mirage/ocaml-cohttp/issues/459 *)
+(* https://github.com/0install/0install/blob/6c0f5c51bc099370a367102e48723a42cd352b3b/ocaml/zeroinstall/http.cohttp.ml#L232-L256 *)
+
+(* What do we have to do to support a proxy? *)
+(* On Native OCaml (not js), the certificate MUST be in a path listed in
+   https://github.com/mirage/ca-certs/blob/fa4ff942f1ac980e2502a0783ef10ade5ba497f2/lib/ca_certs.ml#L34-L50 *)
+(* Or set SSL_CERT_FILE=path/to/cert. But note setting this means no default certs will be included *)
+(* HTTP(s)_PROXY MUST have a scheme, http:// or https://, and must have a port
+   if it isn't on port 80/443 *)
+(* So HTTP_PROXY=http://localhost:8080 or
+   HTTPS_PROXY=https://localhost:8080 *)
+(* And note that HTTP_PROXY is only used for HTTP requests! We almost
+   exclusively use HTTPS urls so make sure both are set*)
+let get_proxy uri =
+  let proxy_uri_env =
+    match Uri.scheme uri with
+    (* TODO support all cases *)
```

Comment:
This is non-obvious as meaning various capitalizations, so I think this can be more verbose.